auth/client/iam/openid4vp.go

```diff
@@ -133,14 +133,32 @@ func (c *OpenID4VPClient) AuthorizationServerMetadata(ctx context.Context, webdi
 	return metadata, nil
 }
 
-func (c *OpenID4VPClient) RequestObject(ctx context.Context, requestURI string) (string, error) {
+func (c *OpenID4VPClient) RequestObject(ctx context.Context, requestURI, requestURIMethod string, walletMetadata *oauth.AuthorizationServerMetadata) (string, error) {
 	iamClient := c.httpClient
 	parsedURL, err := core.ParsePublicURL(requestURI, c.strictMode)
 	if err != nil {
 		return "", fmt.Errorf("invalid request_uri: %w", err)
 	}
-	// the wallet/client acts as authorization server
-	requestObject, err := iamClient.RequestObject(ctx, parsedURL.String())
+
+	var requestObject string
+	switch requestURIMethod {
+	case "", "get":
+		// the wallet/client acts as authorization server
+		requestObject, err = iamClient.RequestObject(ctx, parsedURL.String())
+	case "post":
+		// TODO: consider adding a 'wallet_nonce'
+		form := url.Values{}
+		if walletMetadata != nil {
+			metadataBytes, err := json.Marshal(*walletMetadata)
+			if err != nil {
+				return "", fmt.Errorf("failed to serialize wallet_metadata: %w", err)
```

Comment:
impossible to occur, only if the struct contains a function ref (but it only contains a single string field), just ignore error